src/store/selectors/subjectPicker.js

```diff
@@ -0,0 +1,23 @@
+import getSubjectScore from '../../utils/getSubjectScore'
+
+const subjectPicker = (subjects, subjectsType) => {
+
+  const subjectsArray = subjects.filter(subject => subject.active)
+
+  const maxScoreWrapper = {
+    subject: {},
+    computedScore: 0,
+    formula: '',
+    isEmpty: !subjectsArray.length
+  }
+
+  return subjectsArray.reduce((currentMaxWrapper, subject) => {
+    const currentScoreObj = getSubjectScore(subject, subjectsType)
+
+    return (currentScoreObj.computedScore > currentMaxWrapper.computedScore)
+      ? { subject, ...currentScoreObj, isEmpty: false }
```

Comment:
You can ommit `isEmpty` property if you destructure currentMaxWrapper on the beginning.
`{ ...currentMaxWrapper, subject, ...currentScoreObj }`